src/transport_performance/gtfs/validation.py

```diff
@@ -601,3 +655,803 @@ def get_route_modes(self):
+            Kwargs to pass to fig.update_layout() for
+            additional plot customisation,
+            by default {}
+        return_html : bool, optional
+            Whether or not to return a html string,
+            by default False
+        save_html : bool, optional
+            Whether or not to save the plot as a html file,
+            by default False
+        save_image : bool, optional
+            Whether or not to save the plot as a PNG,
+            by default False
+        out_dir : Union[pathlib.Path, str], optional
+            The directory to save the plot into. If a file extension is added
+            to this directory, it won't be cleaned. Whatever is passed as the
+            out dir will be used as the parent directory of the save, leaving
+            the responsibility on the user to specify the correct path.,
+            by default os.path.join("outputs", "gtfs")
+        img_type : str, optional
+            The type of the image to be saved. E.g, .svg or .jpeg.,
+            by defauly "png"
+
+        Returns
+        -------
+        Union[PlotlyFigure, str]
+            Returns either a HTML string or the plotly figure
+
+        Raises
+        ------
+        ValueError
+            An error is raised if orientation is not 'v' or 'h'.
+        ValueError
+            An error is raised if an invalid iamge type is passed.
+
+        """
+        # parameter type defences
+        _type_defence(summary_df, "summary_df", pd.DataFrame)
+        _type_defence(day_column, "day_column", str)
+        _type_defence(target_column, "target_column", str)
+        _type_defence(plotly_kwargs, "plotly_kwargs", dict)
+        _type_defence(return_html, "return_html", bool)
+        _type_defence(width, "width", int)
+        _type_defence(height, "height", int)
+        _type_defence(xlabel, "xlabel", (str, type(None)))
+        _type_defence(ylabel, "ylabel", (str, type(None)))
+        _type_defence(save_html, "save_html", bool)
+        _type_defence(save_image, "save_iamge", bool)
+        _type_defence(img_type, "img_type", str)
+
+        raw_pth = os.path.join(
+            out_dir,
+            "summary_" + datetime.datetime.now().strftime("%d_%m_%Y-%H_%M_%S"),
+        )
+        _check_parent_dir_exists(raw_pth, "save_pth", create=True)
+
+        # orientation input defences
+        if orientation.lower() not in ["v", "h"]:
+            raise ValueError(
+                "'orientation expected 'v' or 'h'"
+                " (non case sensitive)."
+                f" Found {orientation} of type {type(orientation)}."
+            )
+
+        # dataframe column defences
+        _check_column_in_df(df=summary_df, column_name=target_column)
+        _check_column_in_df(df=summary_df, column_name=day_column)
+
+        # convert column type for better graph plotting
+        summary_df["route_type"] = summary_df["route_type"].astype("object")
+
+        orientation = orientation.lower()
```

Comment:
second time lowering orientation. Good practise to do this but generally safer to get it done at the top of the module so you can rely on that behaviour downstream (and only have to do it the once!)